`TestAuthorizeWithLocksForLocalUser` flakiness

https://github.com/gravitational/teleport/actions/runs/5195772600
https://github.com/gravitational/teleport/actions/runs/5180433207
https://github.com/gravitational/teleport/actions/runs/5179411625
https://github.com/gravitational/teleport/actions/runs/5151590009

The test creates 2 locks and it seems there's a race condition where the second watcher is sometimes created so fast that it receives the event for the first created lock as well:

```
    permissions_test.go:242: 
        	Error Trace:	/__w/teleport/teleport/lib/authz/permissions_test.go:242
        	            				/__w/teleport/teleport/lib/authz/permissions_test.go:176
        	Error:      	Should be empty, but was   &types.LockV2{
        	            	  	Kind:    "lock",
        	            	  	SubKind: "",
make: *** [Makefile:701: test-go-unit] Error 1
        	            	  	Version: "v2",
        	            	  	Metadata: types.Metadata{
        	            	- 		Name: "request-lock",
        	            	+ 		Name: "mfa-lock",
        	            	  		... // 2 ignored and 6 identical fields
        	            	  	},
        	            	  	Spec: types.LockSpecV2{
        	            	  		Target: types.LockTarget{
        	            	  			... // 2 identical fields
        	            	  			Login:          "",
        	            	  			Node:           "",
        	            	- 			MFADevice:      "",
        	            	+ 			MFADevice:      "mfa-device-id",
        	            	  			WindowsDesktop: "",
        	            	- 			AccessRequest:  "test-request",
        	            	+ 			AccessRequest:  "",
        	            	  			Device:         "",
        	            	  			ServerID:       "",
        	            	  			... // 3 identical fields
        	            	  		},
        	            	  		Message: "",
        	            	  		Expires: nil,
        	            	  		... // 5 identical fields
        	            	  	},
        	            	  	XXX_NoUnkeyedLiteral: {},
        	            	  	XXX_unrecognized:     nil,
        	            	  	XXX_sizecache:        0,
        	            	  }
        	Test:       	TestAuthorizeWithLocksForLocalUser
--- FAIL: TestAuthorizeWithLocksForLocalUser (0.03s)
===================================================
```

--- a/lib/authz/permissions_test.go
+++ b/lib/authz/permissions_test.go
@@ -236,15 +236,18 @@ func upsertLockWithPutEvent(ctx context.Context, t *testing.T, client *testClien
 	defer lockWatch.Close()
 
 	require.NoError(t, client.UpsertLock(ctx, lock))
-	select {
-	case event := <-lockWatch.Events():
-		require.Equal(t, types.OpPut, event.Type)
-		require.Empty(t, resourceDiff(lock, event.Resource))
-	case <-lockWatch.Done():
-		t.Fatalf("Watcher exited with error: %v.", lockWatch.Error())
-	case <-time.After(2 * time.Second):
-		t.Fatal("Timeout waiting for lock put.")
-	}
+
+	// Retry a few times to wait for the resource event we expect as the
+	// resource watcher can potentially return events for previously
+	// created resources as well.
+	require.Eventually(t, func() bool {
+		select {
+		case event := <-lockWatch.Events():
+			return types.OpPut == event.Type && resourceDiff(lock, event.Resource) == ""
+		case <-lockWatch.Done():
+			return false
+		}
+	}, 2*time.Second, 100*time.Millisecond)
 }
 
 func TestGetClientUserIsSSO(t *testing.T) {

